x-pack/plugins/security_solution/server/utils/build_validation/route_validation.ts

```diff
@@ -39,3 +40,20 @@ export const buildRouteValidation = <T extends rt.Mixed, A = rt.TypeOf<T>>(
       (validatedInput: A) => validationResult.ok(validatedInput)
     )
   );
+
+export const buildRouteValidationWithExcess = <
+  T extends rt.InterfaceType<rt.Props> | GenericIntersectionC,
+  A = rt.TypeOf<T>
+>(
+  schema: T
+): RouteValidationFunction<A> => (
+  inputValue: unknown,
+  validationResult: RouteValidationResultFactory
+) =>
+  pipe(
+    excess(schema).decode(inputValue),
```

Comment:
I think there is a better name we could use other than just `excess`. Maybe something like `excessPropsValidation` or something that just describes what the `excess` function is really doing behind the scenes which could help make it clearer at first glance without diving into the code.